Userland/Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -404,13 +389,13 @@ static Optional<Vector<Macroblock>> decode_huffman_stream(JPGLoadingContext& con
                 }
             }
 
-            if (!build_macroblocks(context, macroblocks, hcursor, vcursor)) {
+            if (build_macroblocks(context, macroblocks, hcursor, vcursor).is_error()) {
```

Comment:
Same note here about manual error checking / hiding the underlying error. In this case, since there's a bunch of debugging statements below, you can:

```c++
if (auto result = build_macroblocks(context, macroblocks, hcursor, vcursor); result.is_error()) {
    if constexpr (JPG_DEBUG) {
        dbgln("Failed to build Macroblock {}", i);
        dbgln("Huffman stream byte offset {}", context.huffman_stream.byte_offset);
        dbgln("Huffman stream bit offset {}", context.huffman_stream.bit_offset);
    }
    return result.release_error();
}
```